rollout/controller.go

```diff
@@ -275,7 +273,7 @@ func (c *RolloutController) syncHandler(key string) error {
 		return err
 	}
 
-	if rollout.Spec.Paused || isScalingEvent {
+	if len(rollout.Status.PauseConditions) > 0 || isScalingEvent {
```

Comment:
Why don't we honor spec.Paused anymore?

Reply:
I was going to make a separate PR for `.spec.pause` since users-set paused were not ever supported, and this PR is already a larger one. However, I can include it if you want.

Reply:
Separate PR is fine.